lib/SIL/SILFunctionType.cpp

```diff
@@ -231,17 +231,19 @@ CanSILFunctionType SILFunctionType::getAutoDiffAssociatedFunctionType(
   case AutoDiffAssociatedFunctionKind::JVP: {
     SmallVector<SILParameterInfo, 8> differentialParams;
     for (auto &param : wrtParams) {
+      auto paramTan =
+          param.getType()->getAutoDiffAssociatedTangentSpace(lookupConformance);
+      assert(paramTan && "Parameter type does not have a tangent space?");
       differentialParams.push_back(
-          {param.getType()->getAutoDiffAssociatedTangentSpace(lookupConformance)
-              ->getCanonicalType(),
-           param.getConvention()});
+          {paramTan->getCanonicalType(), param.getConvention()});
     }
     SmallVector<SILResultInfo, 8> differentialResults;
     auto &result = getResults()[resultIndex];
+    auto resultTan =
+         result.getType()->getAutoDiffAssociatedTangentSpace(lookupConformance);
```

Comment:
```suggestion
        result.getType()->getAutoDiffAssociatedTangentSpace(lookupConformance);
```

Reply:
I'll add it to my next patch since this PR is urgent.